Sources/SourceControl/GitRepository.swift

```diff
@@ -210,6 +210,11 @@ public struct GitRepositoryProvider: RepositoryProvider, Cancellable {
         return result == ".git" || result == "." || result == directory.pathString
     }
 
+    public func isValidDirectory(_ directory: Basics.AbsolutePath, for repository: RepositorySpecifier) throws -> Bool {
+        let remoteURL = try self.git.run(["-C", directory.pathString, "remote", "get-url", "origin"])
```

Comment:
will the remote always be `origin`? (just wondering)

Reply:
yes iiuc, in this case it us the one that SwiftPM sets